src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -228,6 +234,20 @@ protected internal int GetInvocationCounter()
             return counter;
         }
 
+        /// <summary>
+        /// The implementation of System.Runtime.GetRandom.
+        /// Gets the random number genrated form the VRF
+        /// Primary generates this random number with `prevHash`, the hash of the previous (validators.Length/3 + 1) block
+        /// </summary>
+        /// <returns>The last four bytes of the random number.</returns>
+        protected internal ulong GetRandom()
+        {
+            if (nextNonce == 0) throw new InvalidOperationException("Require persistingBlock");
+            var nonce = nextNonce;
+            nextNonce = BitConverter.ToUInt64(Cryptography.Helper.Sha256(BitConverter.GetBytes(nonce))[..8]);
```

Comment:
Maybe we can use Murmur3 here? It is much faster than sha256.

Reply:
> Maybe we can use Murmur3 here? It is much faster than sha256.
Murmur of neo only has 32 bits results, if we want a bigger random space, we would have to support Murmur-128.

Reply:
Murmur-128 is good.